cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/constraint/AbstractConstraint.java

```diff
@@ -24,13 +24,15 @@
 /**
  * A constraint that is checked before executing a schedule.
  */
-public class ConstraintChecker {
+public abstract class AbstractConstraint implements Constraint {
```

Comment:
In general, interface is for declaring capability, while class is for hierarchy. E.g. you can have `abstract class Constraint` in API, which represents hierarchy of constraints. Some Constraints (although all in this particular case) is "checkable".